tiledb/libtiledb.pyx

```diff
@@ -3697,6 +3697,16 @@ cdef class DenseArrayImpl(Array):
         cdef list values = list()
         cdef tiledb_ctx_t* ctx_ptr = safe_ctx_ptr(self.ctx)
 
+        if isinstance(val, np.ndarray):
+            try:
+                subarray_shape = tuple(int(dim[1]-dim[0]+1) for dim in subarray)
+                np.broadcast_shapes(subarray_shape, val.shape)
+            except:
```

Comment:
Nits:
- move `subarray_shape = ...` before the try/except
- replace bare except with `except ValueError`